exporter_test.go

```diff
@@ -53,6 +53,45 @@ func TestNegativeCounter(t *testing.T) {
 	ex.Listen(events)
 }
 
+// TestDropUnmapped validates that the dropUnmapped global will prevent unmapped
+// metrics from being recorded when set true and that they are recorded when set
+// false.
+func TestDropUnmapped(t *testing.T) {
+	dropUnmapped = func() *bool { b := true; return &b }()
+	events := make(chan Events, 1)
+	c := Events{
+		&CounterEvent{
+			metricName: "foo",
+			value:      1,
+		},
+	}
+	events <- c
+	ex := NewExporter(&metricMapper{}, true)
+
+	// Close channel to signify we are done with the listener after a short period.
+	go func() {
+		time.Sleep(time.Millisecond * 100)
+		close(events)
+	}()
+
+	ex.Listen(events)
+	if len(ex.Counters.Elements) != 0 {
+		t.Fatalf("Unmapped metric recorded inspite of dropUnmapped being set true")
+	}
+	dropUnmapped = func() *bool { b := false; return &b }()
```

Comment:
The fact that you have to jump through such hoops and introduce side effects in a test function by changing the behavior of a global variable is a strong indicator to avoid global variables altogether.